packages/block-library/src/template-part/edit/index.js

```diff
@@ -67,11 +71,29 @@ function ReplaceButton( {
 	);
 }
 
+function TemplatesList( { availableTemplates, onSelect } ) {
+	const shownTemplates = useAsyncList( availableTemplates );
+
+	if ( ! availableTemplates || availableTemplates?.length < 2 ) {
```

Comment:
We should be showing this when the theme only has one pattern because we don't know if a theme may have just one that is NOT the one that is initially on the template part. I'd rather have some themes showing the original pattern if it means we don't miss any patterns on themes that do it differently

Reply:
done in [fc12186](https://github.com/WordPress/gutenberg/pull/55128/commits/fc12186e14c6c427f589243b0cae8c4286b3810b)